onnxruntime/core/providers/cpu/math/matmul.cc

```diff
@@ -169,25 +150,30 @@ Status MatMul<float>::Compute(OpKernelContext* ctx) const {
   size_t max_len = helper.OutputOffsets().size();
   for (size_t i = 0; i < max_len; i++) {
     if (packed_b_) {
-      MlasGemm(CblasNoTrans,
-               static_cast<size_t>(helper.M()),
-               static_cast<size_t>(helper.N()),
-               static_cast<size_t>(helper.K()),
-               1.0f,
-               a_data + helper.LeftOffsets()[i],
-               static_cast<size_t>(helper.K()),
-               packed_b_.get(),
-               0.0f,
-               y_data + helper.OutputOffsets()[i],
-               static_cast<size_t>(helper.N()),
-               thread_pool);
+      MlasGemm(
+          trans_a ? CblasTrans : CblasNoTrans,
```

Comment:
@tracysh Can you verify this is right? These 2 attributes (trans_a, trans_b, alpha) are added so that we can fuse Matmul and TransposeScaleMatmul kernels. Do I need to add error checks around packed_b_ and these 3 attributes meaning when packaed_b_ is true these 3 can only contain default values?

Reply:
This part looks okay, but you'll also need to update MatMul<float>::PrePack to handle the trans_b logic. The call to MlasGemmPackB should use appropriate CblasTrans/CblasNoTrans for the weight tensor. The values for N and K in PrePack should also swap if the tensor is transposed. MlasGemmPackB also takes a ldb parameter as the 5th parameter: this needs to be set to the value of b_shape_[1] instead of always being N as the tensor could be transposed.

Reply:
For the PrePack changes, I had updated matmul_test.cc to include tests where B is and is not a constant in order to exercise the various paths. The TransposeScaleMatMul should be updated to do the same for float. I'll note that MlasGemmPackB(CblasTrans) has been tested in the MLAS unit test, but isn't currently exercised by other ops.

(Thanks for fusing these kernels!)

Reply:
Thanks for the detailed explanation! I will make these changes